notebooks/dia_kernel_exploration.ipynb

```diff
@@ -0,0 +1,1336 @@
+    "        )\n",
+    "\n",
+    "        "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "76b9281b-325b-4cd3-ba5d-d843b21424f1",
+   "metadata": {},
+   "source": [
+    "If we want to modify the run... method of the subtraction task itself, we would subclass AlardLuptonSubtractTask and modify the `run...` method.  Here we just pick \"Modified\" to be generic, but if one had a specific name, that would be good too."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9c28a190-9e3e-4f4d-bb84-fb40ff052a67",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "class ModifiedAlardLuptonSubtractConfig(AlardLuptonSubtractConfig):\n",
+    "    pass\n",
+    "\n",
+    "class ModifiedAlardLuptonSubtractTask(AlardLuptonSubtractTask):\n",
+    "    ConfigClass = ModifiedAlardLuptonSubtractConfig\n",
+    "    _DefaultName = \"modifiedAlardLuptonSubtract\"\n",
+    "    \n",
+    "    # `run` calls `runConvolveTemplate` if the template PSF is better\n",
+    "    # Let's develop this example first before covering the `runConvolveScience` function.\n",
+    "    def runConvolveTemplate(self, template, science, selectSources):\n",
+    "        \"\"\"Convolve the template image with a PSF-matching kernel and subtract\n",
+    "        from the science image.\n",
+    "\n",
+    "        Parameters\n",
+    "        ----------\n",
+    "        template : `lsst.afw.image.ExposureF`\n",
+    "            Template exposure, warped to match the science exposure.\n",
+    "        science : `lsst.afw.image.ExposureF`\n",
+    "            Science exposure to subtract from the template.\n",
+    "        selectSources : `lsst.afw.table.SourceCatalog`\n",
+    "            Identified sources on the science exposure. This catalog is used to\n",
+    "            select sources in order to perform the AL PSF matching on stamp\n",
+    "            images around them.\n",
+    "\n",
+    "        Returns\n",
+    "        -------\n",
+    "        results : `lsst.pipe.base.Struct`\n",
+    "\n",
+    "            ``difference`` : `lsst.afw.image.ExposureF`\n",
+    "                Result of subtracting template and science.\n",
+    "            ``matchedTemplate`` : `lsst.afw.image.ExposureF`\n",
+    "                Warped and PSF-matched template exposure.\n",
+    "            ``backgroundModel`` : `lsst.afw.math.Function2D`\n",
+    "                Background model that was fit while solving for the PSF-matching kernel\n",
+    "            ``psfMatchingKernel`` : `lsst.afw.math.Kernel`\n",
+    "                Kernel used to PSF-match the template to the science image.\n",
+    "        \"\"\"\n",
+    "        # Just some quick debugging to demonstrate that we are running the modifiedAlardLuptonSubtract\n",
+    "        self.log.info(\"Running Modified Subtraction Task\")\n",
+    "        kernelSources = self.makeKernel.selectKernelSources(template, science,\n",
```

Comment:
It would be helpful to test how the `kernelSources` affects the subtraction results. We might want to know whether  including bright sources can reduce the dipole artifacts.